lib/spack/spack/cmd/buildcache.py

```diff
@@ -495,6 +498,14 @@ def install_tarball(spec, args):
     else:
         tarball = bindist.download_tarball(spec)
         if tarball:
+            if args.sha256:
+                checker = spack.util.crypto.Checker(args.sha256)
+                msg = ('cannot verify checksum for "{0}"'
+                       ' [expected={1}]')
+                msg = msg.format(tarball, args.sha256)
+                assert checker.check(tarball), msg
+                tty.msg('Verified SHA256 checksum of the build cache')
```

Comment:
This should probably be a debug message, we don't want to let Spack get too verbose again.

Reply:
Fixed in ef700bd